hydra-cluster/test/Test/GeneratorSpec.hs

```diff
@@ -25,30 +27,36 @@ import Hydra.Ledger.Cardano.Configuration (
   newLedgerEnv,
   protocolParametersFromJson,
   readJsonFileThrow,
-  shelleyGenesisFromJson,
  )
+import Hydra.Logging (showLogsOnFailure)
 import Test.Aeson.GenericSpecs (roundtripSpecs)
+import Test.EndToEndSpec (withClusterTempDir)
 import Test.QuickCheck (
   Positive (Positive),
   Property,
   counterexample,
   forAll,
   idempotentIOProperty,
+  property,
+  quickCheck,
  )
 
 spec :: Spec
 spec = parallel $ do
   roundtripSpecs (Proxy @Dataset)
-  prop "generates a Dataset that keeps UTXO constant" prop_keepsUTxOConstant
+  around showLogsOnFailure $
+    it "generates a Dataset that keeps UTXO constant" $ \tracer -> do
+      failAfter 60 $
+        withClusterTempDir "queryGenesisParameters" $ \tmpDir -> do
+          withCardanoNodeDevnet tracer tmpDir $ \RunningNode{nodeSocket, networkId} -> do
+            globals <- newGlobals =<< queryGenesisParameters networkId nodeSocket QueryTip
+            quickCheck $ property $ prop_keepsUTxOConstant globals
```

Comment:
Don't run quickcheck here.. property itself must be enough

Reply:
`prop_keepsUTxOConstant` runs in IO now?

Reply:
Ah.. but then we should make the whole a `monadicIO` property. By using `quickcheck` here it will not pick up the command line options given to the hspec test runner.